src/Components/Server/src/Circuits/RemoteNavigationManager.cs

```diff
@@ -131,11 +131,20 @@ protected override void SetNavigationLockState(bool value)
             return;
         }
 
-        SetHasLocationChangingListeners(value);
+        _ = SetHasLocationChangingListenersAsync(value);
     }
 
-    private void SetHasLocationChangingListeners(bool value)
-        => _jsRuntime.InvokeVoidAsync(Interop.SetHasLocationChangingListeners, value).Preserve();
+    private async Task SetHasLocationChangingListenersAsync(bool value)
+    {
+        try
+        {
+            await _jsRuntime.InvokeVoidAsync(Interop.SetHasLocationChangingListeners, value);
+        }
+        catch (JSDisconnectedException)
+        {
+            // If the browser is gone, we don't need it to clean up any browser-side state
```

Comment:
This could also happen during a temporary disconnect, doesn't it?

Reply:
I think `JSDisconnectedException`s are only thrown upon permanent circuit disconnection: https://github.com/dotnet/aspnetcore/blob/3c542fec8e5e47db328f8bd631d2a5e4feb28db4/src/Components/Server/src/Circuits/RemoteJSRuntime.cs#L109-L113